ui/app/AppLayouts/Chat/panels/communities/SortableTokenHoldersPanel.qml

```diff
@@ -0,0 +1,197 @@
+    property var model
+
+    property string tokenName
+    property bool isSelectorMode: false
+
+    readonly property alias sortBy: holdersList.sortBy
+    readonly property alias sorting: holdersList.sorting
+
+    readonly property bool empty: countCheckHelper.count === 0
+
+    signal viewProfileRequested(string address)
+    signal viewMessagesRequested(string address)
+    signal airdropRequested(string address)
+    signal remoteDestructRequested(string address)
+    signal kickRequested(string address)
+    signal banRequested(string address)
+
+    signal generalAirdropRequested
+
+    Instantiator {
+        id: countCheckHelper
+
+        model: root.model
+        delegate: QtObject {}
+    }
+
+    TokenHoldersProxyModel {
+        id: proxyModel
+
+        sourceModel: root.model
+        searchText: searcher.text
+
+        sortBy: holdersList.sortBy
+        sortOrder: holdersList.sorting === SortableTokenHoldersList.Sorting.Descending
+                   ? Qt.DescendingOrder : Qt.AscendingOrder
+    }
+
+    QtObject {
+        id: d
+
+        readonly property int red2Color: 4
+    }
+
+    contentItem: ColumnLayout {
+        anchors.fill: parent
+        spacing: 0
+
+        StatusBaseText {
+            Layout.fillWidth: true
+
+            wrapMode: Text.Wrap
+            font.pixelSize: Style.current.primaryTextFontSize
+            color: Theme.palette.baseColor1
+
+            text: qsTr("%1 token holders").arg(root.tokenName)
+        }
+
+        SearchBox {
+            id: searcher
+
+            Layout.fillWidth: true
+            Layout.topMargin: 12
+
+            visible: !root.empty
+
+            topPadding: 0
+            bottomPadding: 0
+            minimumHeight: 36 // by design
+            maximumHeight: minimumHeight
+
+            placeholderText: qsTr("Search hodlers")
+        }
+
+        StatusBaseText {
+            Layout.fillWidth: true
+            Layout.topMargin: 12
+
+            wrapMode: Text.Wrap
+            font.pixelSize: Style.current.primaryTextFontSize
+            color: Theme.palette.baseColor1
+
+            visible: searcher.text.length > 0
+
+            text: (searcher.text.length > 0 && proxyModel.count > 0)
+                  ? qsTr("Search results") : qsTr("No hodlers found")
+        }
+
+        NoHoldersPanel {
+            Layout.fillWidth: true
+            Layout.topMargin: Style.current.padding
+
+            visible: root.empty
+
+            onAirdropRequested: root.generalAirdropRequested()
+        }
+
+        SortableTokenHoldersList {
+            id: holdersList
+
+            visible: !root.empty && proxyModel.count > 0
+
+            Layout.fillWidth: true
+            Layout.preferredHeight: contentHeight
+            Layout.topMargin: 20
+
+            model: proxyModel
+
+            onClicked: {
+                if (mouse.button !== Qt.RightButton)
+                    return
```

Comment:
Wondering if here we should do some action too.

Reply:
According to figma, it's right-click menu: https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=29673-619998&t=Rw4edL9LegNa9ez6-0